### ISSUE TITLE
add support for data.table and table

### ISSUE THREAD

**Issue by timelyportfolio** [smartinsightsfromdata/rpivotTable#5]
As of now, only `data.frame` works.  I'll work on adding conversion for `data.table` and `table` into the function.

**Comment by smartinsightsfromdata** [comment]
@timelyportfolio   This is not correct.  I am using rpivotable in a large shiny project I'm working on and I use it exclusively with data.table.  I'll check on table.

**Comment by timelyportfolio** [comment]
Well, it did not work on my end.  I'll just make it explicit, so it will not be a problem.

**Comment by smartinsightsfromdata** [comment]
@timelyportfolio Please try again: I just tested it now.

**Comment by timelyportfolio** [comment]
I believe it depends on the `toJSON` that gets called and the data in the `data.table`.  Not sure why being explicit will cause a problem.

**Comment by smartinsightsfromdata** [comment]
@timelyportfolio there is *no* toJSON in the actual code.  I used in the example to quickly transform some of the json I got hacking a web example that Nicolas was using on Canadian election.

**Comment by timelyportfolio** [comment]
`htmlwidgets` converts all parameters to JSON [here](https://github.com/ramnathv/htmlwidgets/blob/master/R/htmlwidgets.R#L127) .  Since you don't believe me, try this:

```
rpivotTable(data.table(iris))
```

**Comment by smartinsightsfromdata** [comment]
@timelyportfolio xtabs and table is different. But I am not sure it is a good idea to make the code too complex to manage these types.  It is much better (and robust) to ask the user to convert.
Another story could be the new type tbl used by Hadley in dplyr, that he wants to design in compatibility with lap tools.  I will ask him what are his plans and report back.

**Comment by timelyportfolio** [comment]
`tbl_df` will work fine with my change.  Can add a warning about conversion so user is not surprised.

**Comment by smartinsightsfromdata** [comment]
@timelyportfolio  since you do not believe *me* try this in dat01.R (in the example folder):
```R
rpivotTable(data = as.data.table(jsdf_))
```
I will look into why iris does not work, but it certainly *not* because it is a data.table.

**Comment by timelyportfolio** [comment]
I never said your data did not work.  What I said is that I have a case where an error occurs and that I think we should handle.  See these type checks/conversions from RStudio.

https://github.com/rstudio/dygraphs/blob/master/R/dygraph.R#L31-L33
https://github.com/rstudio/DT/blob/master/R/datatables.R#L43-L53

**Comment by smartinsightsfromdata** [comment]
@timelyportfolio the error with iris occurs because iris column names have a dot and I suspect pivotable does not like that.  If you convert the dot in the name into a _ (underscore) it works.  Try it.

**Comment by timelyportfolio** [comment]
I think this is another expectation that gets solved with a simple conversion using `data.table` built in conversion.  `toJSON` was not built with `data.table` in mind so result will be more accurate with `data.frame`.

I'll submit a pull and you can decide.

**Comment by smartinsightsfromdata** [comment]
@timelyportfolio it seems that the problem is with the dot but this emerges only after a conversion from data.frame to data.table.  (a data.frame with no dots in the colnames is OK and of course a native data.table is OK).
As I am a strong data.table supporter / user, I'll report it to the data.table project: if there is something that breaks the data.frame / data.table parallelism Matt D. and co. needs to have a look at it.

**Comment by smartinsightsfromdata** [comment]
@timelyportfolio I think the issue is between RJSONIO and data.table. See [here](http://stackoverflow.com/questions/28372298/json-issue-bug-with-data-table-and-rjsonio-jsonlite)

**Comment by timelyportfolio** [comment]
Oh, I know it is.  In no way should this discussion suggest that I do not support `data.table`. I simply want `rpivotTable` to be robust.  Once `data.table` is converted to `JSON` it doesn't matter what form it is in. In fact, `toJSON` will force convert unknown classes anyways.  The step I added just assures that users get a good result.
